server/src/api/v1.py

```diff
@@ -652,3 +656,45 @@ def queue_wait_time_percentiles_get():
             queue["wait_times"]
         )
     return queue_percentile_data
+
+
+def generate_token(permissions, secret_key):
+    """Generates JWT token with queue permission given a secret key"""
+    expiration_time = datetime.utcnow() + timedelta(seconds=2)
+    token_payload = {
+        "exp": expiration_time,
+        "iat": datetime.now(timezone.utc),  # Issued at time
+        "sub": "access_token",
+        "permissions": permissions,
+    }
+    token = jwt.encode(token_payload, secret_key, algorithm="HS256")
+    return token
+
+
+def validate_client_key_pair(client_id: str, client_key: str):
+    """
+    Checks client_id and key pair for validity and returns their permissions
+    """
+    client_permissions_entry = database.mongo.db.client_permissions.find_one(
+        {
+            "client_id": client_id,
+            "client_secret_hash": client_key,
+        }
+    )
+    if client_permissions_entry is None:
+        return {}
+
+    permissions = client_permissions_entry["permissions"]
+    return permissions
+
+
+SECRET_KEY = secrets.token_hex(20)
+
+
+@v1.get("/authenticate/token/<client_id>")
```

Comment:
What is the idea behind the use of "token" in `authenticate/token`? I might be missing something but if we are authenticating a client or a user, this could be `/authenticate/client` or `authenticate/user`. Which would also be compatible with adding something like `/authenticate/agent` later.

Reply:
I was just copying the oauth standard endpoint of `/oauth/token` - https://auth0.com/docs/authenticate/protocols/oauth